src/main/java/meteordevelopment/meteorclient/mixin/MinecraftClientMixin.java

```diff
@@ -116,9 +121,17 @@ private void onTick(CallbackInfo info) {
         Profilers.get().pop();
     }
 
-    @Inject(method = "doAttack", at = @At("HEAD"))
+    @Inject(method = "doAttack", at = @At("HEAD"), cancellable = true)
     private void onAttack(CallbackInfoReturnable<Boolean> cir) {
         CPSUtils.onAttack();
+
+        NoInteract noInteract = Modules.get().get(NoInteract.class);
+
+        if (noInteract.isActive() && noInteract.shouldCancelMissedAttacks() &&
+            crosshairTarget != null && crosshairTarget.getType() == HitResult.Type.MISS) {
+            cir.setReturnValue(false);
+            player.swingHand(Hand.MAIN_HAND, false);
```

Comment:
```suggestion
            player.swingHand(Hand.MAIN_HAND, false);
            cir.setReturnValue(false);
```
(`setReturnValue` immediately exits the function)